packages/broker/src/storage/StoragePlugin.ts

```diff
@@ -0,0 +1,91 @@
+import { router as dataQueryEndpoints } from './DataQueryEndpoints'
+import { router as dataMetadataEndpoint } from './DataMetadataEndpoints'
+import { router as storageConfigEndpoints } from './StorageConfigEndpoints'
+import { Plugin, PluginOptions } from '../Plugin'
+import { StreamFetcher } from '../StreamFetcher'
+import { Storage, startCassandraStorage } from './Storage'
+import { StorageConfig } from './StorageConfig'
+import { StreamPart } from '../types'
+import { Wallet } from 'ethers'
+import PLUGIN_CONFIG_SCHEMA from './config.schema.json'
+
+export interface StoragePluginConfig {
+    cassandra: {
+        hosts: string[],
+        username: string
+        password: string
+        keyspace: string,
+        datacenter: string
+    }
+    storageConfig: {
+        refreshInterval: number
+    }
+}
+
+export class StoragePlugin extends Plugin<StoragePluginConfig> {
+
+    private cassandra: Storage|undefined
+    private storageConfig: StorageConfig|undefined
+
+    constructor(options: PluginOptions) {
+        super(options)
+    }
+
+    async start() {
```

Comment:
I feel like this should have some sort of re-entry protection to prevent running `start` twice, or while either `start` or `stop` are still pending, etc. Might not be possible to do this currently, but should this ever happen it will produce some weird behaviour and might not be easy to spot what the problem is.

Reply:
One way to achieve what Tim described is to
- Make this class into an abstract base class
- Add a boolean field `this.started = false`
- Modify `start` method to work as follows
```ts
async start() {
    if (this.started) {
        throw new Error('started already')
    }
    this.started = true
    this.doStart()
}
```
- Re-name and change the visibility of method `async start()` to `abstract protected doStart()`, which the extending class would then implement.

Reply:
Added documentation to `Plugin.ts` about `start` and `stop` lifecycle methods. Plugins can therefore now assume that the methods are called in valid order, as the behaviour is documented. 

I'd say that there is no need to to add additional guards to Broker to prevent us from calling lifecycle methods in invalid order. If that happened, it would maybe even be better to catch the bug by crashing the app instead of silently ignoring the error. (Depends of how we want to handle error situations in general. I've understood that this approach is currently used in Broker as we are more in development mode than in maintenance mode).